mmf_cli/run.py

```diff
@@ -108,15 +108,26 @@ def run(opts: typing.Optional[typing.List[str]] = None, predict: bool = False):
         else:
             distributed_main(0, configuration, predict)
     elif config.distributed.world_size > 1:
-        assert config.distributed.world_size <= torch.cuda.device_count()
-        port = random.randint(10000, 20000)
-        config.distributed.init_method = f"tcp://localhost:{port}"
-        config.distributed.rank = None
-        torch.multiprocessing.spawn(
-            fn=distributed_main,
-            args=(configuration, predict),
-            nprocs=config.distributed.world_size,
-        )
+        if is_xla():
+            import torch_xla.distributed.xla_multiprocessing as xmp
+
+            torch.multiprocessing.set_sharing_strategy("file_system")
+            xmp.spawn(
+                fn=distributed_main,
+                args=(configuration, predict),
+                nprocs=8,  # use all 8 TPU cores
```

Comment:
noob question: Does this number always stays fixed or should be configurable? Is there a situation where it needs to be configured?

Reply:
Hey Aman, this branch should be activated only when distributed world size > 1. For TPUs you can either use 1 core or 8 cores or a larger pod slice (32, 128, 256, 512, 1024, 2048). For single core xmp.spawn is not used and main is run directly. For larger size, per host (VM) we will use all 8 cores and hence this number is fixed at 8.